casper/src/test/scala/coop/rchain/casper/MergeabilityRules.scala

```diff
@@ -446,19 +496,67 @@ trait MergeabilityRules {
     "(C!) (C!)" -> PersistentNoMatch(C_, S0)(C_, S0)(Nil)
   )
 
+  val peekMergeabilityCases = List(
+    "!4 P!"     -> HadItsMatch(S0)(P1)(F0, S1),
+    "!4 (P!)"   -> VolatileEvent(S0)(P1, S1)(F0),
+    "!X PX"     -> IncomingCouldMatch(S0)(P_)(Nil),
+    "!4 PX"     -> HadItsMatch(S0)(P_)(F0),
+    "!C PX"     -> HadItsMatch(S0)(P_)(C_),
+    "!X (P!)"   -> PeekedNoMatch(S0)(P1, S1)(Nil),
+    "!X P!"     -> HadItsMatch(S0)(P_)(S1),
+    "(!4) P!"   -> VolatileEvent(S0, F0)(P_)(S1),
+    "(!4) PX"   -> VolatileEvent(S0, F0)(P_)(Nil),
+    "(!C) P!"   -> PersistentCouldMatch(S0, C_)(P_)(S0),
+    "(!C) PX"   -> PersistentNoMatch(S0, C_)(P_)(Nil),
+    "(4!) (P!)" -> VolatileEvent(F_, S0)(P_, S0)(Nil),
+    "(4!) P!"   -> VolatileEvent(F0, S0)(P_)(S1),
+    "(4!) PX"   -> VolatileEvent(F0, S0)(P_)(Nil),
+    "(P!) (P!)" -> VolatileEvent(P0, S0)(P0, S0)(Nil),
+    "(P!) P!"   -> PeekedNoMatch(P0, S0)(P_)(S1),
+    "4! (P!)"   -> PeekedNoMatch(F0)(P1, S1)(S0),
+    // The case below can merge iff peek block is executed as first - conflicts for now
+    "4! P!"     -> CouldMatchSameConflicts(F0)(P_)(S0),
+    "4! P!"     -> CouldMatchSameMerges(F0)(P1)(S0, S1),
+    "4! PX"     -> SamePolarityMerge(F0)(P1)(S0),
+    "4X PX"     -> SamePolarityMerge(F_)(P_)(Nil),
+    "P! (C!)"   -> PersistentNoMatch(P_)(C0, S0)(S1),
+    "P! (P!)"   -> HadItsMatch(S0)(P_)(F0),
+    "P! C!"     -> CouldMatchSameConflicts(P_)(C_)(S0),
+    "P! P!"     -> CouldMatchSameMerges(P0)(P1)(S0, S1),
+    "PX (C!)"   -> PersistentNoMatch(P0)(C_, S0)(Nil),
+    "PX (P!)"   -> PeekedCouldMatch(P0)(P_, S0)(Nil),
+    "PX (P!)"   -> PeekedCouldMatch(P_, S0)(P0)(Nil),
+    "PX C!"     -> SamePolarityMerge(P0)(C_)(S1),
+    "PX CX"     -> SamePolarityMerge(P_)(C_)(Nil),
+    "PX PX"     -> SamePolarityMerge(P_)(P_)(Nil),
+    "!C P!"     -> HadItsMatch(S0)(P_)(C0, S1),
+    "(!4) (P!)" -> VolatileEvent(S0, F_)(P_, S0)(Nil),
+    "!C (P!)"   -> PeekedNoMatch(S0)(P1, S1)(C0),
+    "(P!) (C!)" -> PeekedCouldMatch(P0, S0)(C_, S0)(Nil),
+    "(!C) (P!)" -> PeekedCouldMatch(S0, C_)(P_, S0)(Nil),
+    "(P!) C!"   -> PeekedCouldMatch(P0, S0)(C0)(S1),
+    "(P!) CX"   -> PeekedCouldMatch(P_, S0)(C_)(Nil),
+    "4X (P!)"   -> PeekedCouldMatch(F0)(P_, S0)(Nil),
+    "P! CX"     -> SamePolarityMerge(P0)(C1)(S0),
+    "4X P!"     -> SamePolarityMerge(F0)(P_)(S1),
+    "PX P!"     -> SamePolarityMerge(P0)(P_)(S1),
+    // TODO:
+    "P! P!" -> CouldMatchSameConflicts(P_)(P_)(S0) // conflicts until replay errors are fixed
```

Comment:
Please ping me so that we can discuss the problem and fix approach - this will likely make the PR and its review easier :). So far, I'm not sure why removing sequence numbers will fix that.

Reply:
I see that you are still "requesting changes". What exactly are the changes that are absolutely required before this PR can be merged?